### PR TITLE
Fix mapping of files from bundle to only map the necessary part

### LINKED ISSUE #41816: .NET 5 Preview 8, Single-File deployment of WebApi project throws Segmentation fault on ARM
### Description

Repro:

1. Create new WebApi project
dotnet new webapi -n RaspSegFaultTest

2. Publish as single file 
dotnet publish -c Release -r linux-arm /p:PublishSingleFile=true

3. Try to run on ARM (Raspberry PI 3 in my case)
Result:
```
Unhandled exception. Segmentation fault
```
### Configuration

λ dotnet --info                                                           
.NET SDK (reflecting any global.json):                                    
 Version:   5.0.100-preview.8.20417.9                                     
 Commit:    fc62663a35                                                    
                                                                          
Runtime Environment:                                                      
 OS Name:     Windows                                                     
 OS Version:  10.0.17763                                                  
 OS Platform: Windows                                                     
 RID:         win10-x64                                                   
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100-preview.8.20417.9\      
                                                                          
Host (useful for support):                                                
  Version: 5.0.0-preview.8.20407.11                                       
  Commit:  bf456654f9                                                     

ARM env:

Raspberry PI 3 model 3

```
uname -a
Linux nextcloudplus 4.19.66-v7+ #1253 SMP Thu Aug 15 11:49:46 BST 2019 armv7l GNU/Linux
```

### PR DESCRIPTION
**Pull request by vitek-karas**
Mapping the whole bundle every time leads to VM space starvation. Which mostly matters on 32bit.

The fix is to map just the part of the bundle file which contains the assembly we're after. Memory mapping API needs the offset to be aligned to allocation granularity, so a bit more math to prealign and so on, but otherwise straightforward.

Fixes https://github.com/dotnet/runtime/issues/41816

### CODE REVIEW

**Comment by vitek-karas** [comment]
Tested on win-x64, win-x86 and linux-arm.

**Review by LakshanF — APPROVED** [review]

**Review by jkotas — APPROVED** [review]

**Review by sbomer — APPROVED** [review]

File: src/coreclr/src/vm/peimagelayout.cpp [hunk 1]
```diff
@@ -712,14 +712,20 @@ FlatImageLayout::FlatImageLayout(PEImage* pOwner)
         if (m_FileMap == NULL)
             ThrowLastError();
 
-        //DWORD lowPart = (DWORD)offset;
-        //DWORD highPart = (DWORD)(offset >> 32);
-        char *addr = (char*)CLRMapViewOfFile(m_FileMap, FILE_MAP_READ, 0, 0, 0);
-        addr += offset;
-        m_FileView.Assign((LPVOID)addr);
+        UINT32 alignment = g_SystemInfo.dwAllocationGranularity;
```
[the remark this answers is not in the document]
Comment (vitek-karas) [reply src/coreclr/src/vm/peimagelayout.cpp:717]: It's a requirement of the memory mapping API.
* Windows - must be allocation granularity aligned (which is typically 64KB on "normal" Windows)
* Linux - must be page aligned (PAL sets allocation granularity to page size)

**Review by agocke — APPROVED** [review]
LGTM, thanks!

**Review by VSadov — APPROVED** [review]

**Pull request merged**